array/array.go

```diff
@@ -0,0 +1,131 @@
+package array
+
+import (
+	"errors"
+)
+
+type Array struct {
+	arr  []int
+	size int
+}
+
+func NewArray(capacity int) *Array {
+	return &Array{
+		arr: make([]int, capacity),
+	}
+}
+
+func (a *Array) Size() int {
+	return a.size
+}
+
+func (a *Array) Capacity() int {
+	return len(a.arr)
```

Comment:
Maybe we should use builtin function `cap`

Reply:
I don't understand the differences between `cap` and `len` when calling them on an array yet.